version: "3.5"
services:
  elasticsearch:
    image: docker.elastic.co/elasticsearch/elasticsearch:8.6.2
    container_name: elasticsearch
    volumes:
      - elasticsearch-data:/usr/share/elasticsearch/data
    ports:
      - "9200:9200"
    networks:
      - sofia  
    restart: always
    environment:
      - xpack.security.enabled=false
      - discovery.type=single-node
    ulimits:
      memlock:
        soft: -1
        hard: -1
  kibana:
    depends_on:
      - elasticsearch
    image: docker.elastic.co/kibana/kibana:8.6.2
    container_name: kibana
    volumes:
      - kibana-data:/usr/share/kibana/data
    ports:
     - "5601:5601"
    networks:
      - sofia
    restart: always
    environment:
      - ELASTICSEARCH_HOSTS=http://elasticsearch:9200
  logstash:
    build:
      context: .
      dockerfile: Dockerfile-logstash
    container_name: sem_logstash
    # restart: on-failure
    depends_on:
      - elasticsearch
      - sofiadb
    volumes:
      - ./volumes/logstash/pipeline/:/usr/share/logstash/pipeline/
      - ./volumes/logstash/config/logstash.yml:/usr/share/logstash/config/logstash.yml
      - ./volumes/logstash/config/pipelines.yml:/usr/share/logstash/config/pipelines.yml
      - ./volumes/logstash/config/queries/:/usr/share/logstash/config/queries/
    networks:
      - sofia
    logging:
      driver: "json-file"
      options:
        max-size: "10k"
        max-file: "10"
volumes:
  elasticsearch-data:
    driver: local
  kibana-data:
    driver: local
networks:
  sofia:
    name: sofia
    external: true
    driver: bridge